core/client/db/src/lib.rs

```diff
@@ -569,7 +564,7 @@ impl<Block> client::backend::Backend<Block, Blake2Hasher> for Backend<Block> whe
 						BlockId::Hash(parent_hash),
 					)?;
 
-					// update block number to hash lookup entries.
+					// uncanonicalize
```

Comment:
Probably extract `for` in a separate `fn`? Seems like difference of for-s is only the `prev_lookup_key` && `new_lookup_key` + returning `NotInFinalizedChain` err in `retracted` for?

Reply:
@snd maybe address this in a followup PR?